agent/agent-gc-monitor/gc-monitor-tests/src/test/java/com/microsoft/gcmonitortests/VariousCollectorsTest.java

```diff
@@ -22,96 +22,68 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.StringJoiner;
 import java.util.function.Predicate;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledForJreRange;
+import org.junit.jupiter.api.condition.EnabledOnOs;
 
 class VariousCollectorsTest {
 
   @Test
-  void testCms() throws IOException, UnableToMonitorMemoryException, InterruptedException {
-    try {
-      List<GcCollectionEvent> events =
-          new GcProcessRunner("-XX:+UseConcMarkSweepGC", 50).getGcCollectionEvents();
-
-      assetGcsArePresent(events);
-    } catch (GcNotPresentException e) {
-      // CMS deprecated
-    }
+  @EnabledForJreRange(max = JAVA_13)
+  void testCms() throws Exception {
+    testGc("-XX:+UseConcMarkSweepGC", 50);
   }
 
-  private static void assetGcsArePresent(List<GcCollectionEvent> events) {
-    assertThat(youngGcIsPresent(events)).isTrue();
-    assertThat(tenuredGcIsPresent(events)).isTrue();
-    assertThat(systemGcIsPresent(events)).isTrue();
+  @Test
+  void testParallel() throws Exception {
+    testGc("-XX:+UseParallelGC", 70);
   }
 
   @Test
-  void testParallel()
-      throws IOException, UnableToMonitorMemoryException, InterruptedException,
-          GcNotPresentException {
-    List<GcCollectionEvent> events =
-        new GcProcessRunner("-XX:+UseParallelGC", 70).getGcCollectionEvents();
-
-    print(events);
-
-    assetGcsArePresent(events);
+  void testG1() throws Exception {
+    testGc("-XX:+UseG1GC", 50);
   }
 
   @Test
-  void testG1()
-      throws IOException, UnableToMonitorMemoryException, InterruptedException,
-          GcNotPresentException {
-    List<GcCollectionEvent> events =
-        new GcProcessRunner("-XX:+UseG1GC", 50).getGcCollectionEvents();
-
-    assetGcsArePresent(events);
+  void testSerial() throws Exception {
+    testGc("-XX:+UseSerialGC", 50);
   }
 
   @Test
-  void testSerial() throws IOException, UnableToMonitorMemoryException, InterruptedException {
-    try {
-      List<GcCollectionEvent> events =
-          new GcProcessRunner("-XX:+UseSerialGC", 50).getGcCollectionEvents();
-      assetGcsArePresent(events);
-    } catch (GcNotPresentException e) {
-      // to be expected for some time
-    }
+  @EnabledForJreRange(min = JAVA_11)
+  void testShenandoah() throws Exception {
+    testGc("-XX:+UseShenandoahGC", 50);
   }
 
-  // TODO: Enable when this can use Java 11
-  @Disabled
+  // TODO (trask) failing on Java 17 due to "Could not find factory for ZGC Cycles"
   @Test
-  void testShenandoah() throws IOException, UnableToMonitorMemoryException, InterruptedException {
-    try {
-      List<GcCollectionEvent> events =
-          new GcProcessRunner("-XX:+UseShenandoahGC", 50).getGcCollectionEvents();
-
-      assetGcsArePresent(events);
-    } catch (GcNotPresentException e) {
-      // to be expected for some time
-    }
+  @EnabledForJreRange(min = JAVA_11, max = JAVA_11)
```

Comment:
@EnabledForJreRange(min = JAVA_11)?

Reply:
for some reason its failing on Java 17, see TODO comment above